File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        n1=len(word1)
        n2=len(word2)
        merged=""

        for i in range (0,min(n1,n2)):

            merged=merged+word1[i]+word2[i]

        if n1>n2:
            merged=merged+word1[n2:]
        else:
            merged=merged+word2[n1:]

        return merged

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {

        //abcd pq

    //abc pqr
       
        StringBuilder result= new StringBuilder();

        int minLen=Math.min(word1.length(),word2.length());

        for (int i=0;i<minLen;i++){
            result.append(word1.charAt(i));
            result.append(word2.charAt(i));
        }

        if (word1.length()>minLen){
            result.append(word1.substring(minLen));
        }

        if (word2.length()>minLen){
            result.append(word2.substring(minLen));
        }

       

         return result.toString();
        
    }
}